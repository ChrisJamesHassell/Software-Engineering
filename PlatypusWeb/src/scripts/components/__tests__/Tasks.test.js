import React from 'react';
import ReactDOM from 'react-dom';
import { TaskList } from '../Pages/Tasks.js';
import {shallow} from 'enzyme';

describe('Events', () =>{
    const taskList = shallow(<TaskList />);

    it('should render tasks', () => {
        // console.log(taskList.debug())
        expect(taskList.find('Fragment').exists()).toBe(true);
    });
});