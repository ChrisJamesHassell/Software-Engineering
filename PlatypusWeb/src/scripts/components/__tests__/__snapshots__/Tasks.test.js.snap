// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList TaskList snap matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TaskList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Connect(Droppable)
          direction="vertical"
          droppableId="tasks-high"
          ignoreContainerClipping={false}
          isDropDisabled={false}
          type="DEFAULT"
        >
          [Function]
        </Connect(Droppable)>,
        <Connect(Droppable)
          direction="vertical"
          droppableId="tasks-medium"
          ignoreContainerClipping={false}
          isDropDisabled={false}
          type="DEFAULT"
        >
          [Function]
        </Connect(Droppable)>,
        <Connect(Droppable)
          direction="vertical"
          droppableId="tasks-low"
          ignoreContainerClipping={false}
          isDropDisabled={false}
          type="DEFAULT"
        >
          [Function]
        </Connect(Droppable)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "children": [Function],
          "direction": "vertical",
          "droppableId": "tasks-high",
          "ignoreContainerClipping": false,
          "isDropDisabled": false,
          "type": "DEFAULT",
        },
        "ref": null,
        "rendered": [Function],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "children": [Function],
          "direction": "vertical",
          "droppableId": "tasks-medium",
          "ignoreContainerClipping": false,
          "isDropDisabled": false,
          "type": "DEFAULT",
        },
        "ref": null,
        "rendered": [Function],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "children": [Function],
          "direction": "vertical",
          "droppableId": "tasks-low",
          "ignoreContainerClipping": false,
          "isDropDisabled": false,
          "type": "DEFAULT",
        },
        "ref": null,
        "rendered": [Function],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Connect(Droppable)
            direction="vertical"
            droppableId="tasks-high"
            ignoreContainerClipping={false}
            isDropDisabled={false}
            type="DEFAULT"
          >
            [Function]
          </Connect(Droppable)>,
          <Connect(Droppable)
            direction="vertical"
            droppableId="tasks-medium"
            ignoreContainerClipping={false}
            isDropDisabled={false}
            type="DEFAULT"
          >
            [Function]
          </Connect(Droppable)>,
          <Connect(Droppable)
            direction="vertical"
            droppableId="tasks-low"
            ignoreContainerClipping={false}
            isDropDisabled={false}
            type="DEFAULT"
          >
            [Function]
          </Connect(Droppable)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "children": [Function],
            "direction": "vertical",
            "droppableId": "tasks-high",
            "ignoreContainerClipping": false,
            "isDropDisabled": false,
            "type": "DEFAULT",
          },
          "ref": null,
          "rendered": [Function],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "children": [Function],
            "direction": "vertical",
            "droppableId": "tasks-medium",
            "ignoreContainerClipping": false,
            "isDropDisabled": false,
            "type": "DEFAULT",
          },
          "ref": null,
          "rendered": [Function],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "children": [Function],
            "direction": "vertical",
            "droppableId": "tasks-low",
            "ignoreContainerClipping": false,
            "isDropDisabled": false,
            "type": "DEFAULT",
          },
          "ref": null,
          "rendered": [Function],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
